notebooks/225-yolov7-optimization/225-yolov7-optimiization.ipynb

```diff
@@ -0,0 +1,706 @@
+    "MODEL_LINK = \"https://github.com/WongKinYiu/yolov7/releases/download/v0.1/yolov7-tiny.pt\"\n",
+    "DATA_DIR = Path(\"data/\")\n",
+    "MODEL_DIR = Path(\"model/\")\n",
+    "MODEL_DIR.mkdir(exist_ok=True)\n",
+    "DATA_DIR.mkdir(exist_ok=True)\n",
+    "\n",
+    "download_file(MODEL_LINK, directory=MODEL_DIR, show_progress=True)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Check model inference\n",
+    "\n",
+    "`detect.py` script run pytorch model inference and save image as result,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python detect.py --weights model/yolov7-tiny.pt --conf 0.25 --img-size 640 --source inference/images/horses.jpg\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# visualize prediction result\n",
+    "from PIL import Image\n",
+    "Image.open('runs/detect/exp/horses.jpg')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Export to ONNX"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In order to obtain ONNX model, we will use `export.py` script. Let's check it's arguments."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python export.py - -help\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The most important parameters:\n",
+    "* `--weights` - path to model weigths checkpoint\n",
+    "* `--img-size` - size of input image for onnx tracing\n",
+    "\n",
+    "As ONNX is less flexible format then PyTorch, there is also opportunity to setup configurable parameters for results postprocessing included in model:\n",
+    "* `--end2end` - export full model to onnx including postporcessing\n",
+    "* `--grid` - export Detect layer as part of model\n",
+    "* `--topk-all` - topk elements for all images\n",
+    "* `--iou-thres` - intersection over union threshold for NMS\n",
+    "* `--conf-thres` - minimal confidence threshold\n",
+    "* `--max-wh` - max bounding box width and height for NMS"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python export.py --weights model/yolov7-tiny.pt \\\n",
+    "    --grid --end2end --simplify \\\n",
+    "    --topk-all 100 --iou-thres 0.65 --conf-thres 0.35 \\\n",
+    "    --img-size 640 640 --max-wh 640\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert ONNX Model to OpenVINO Intermideate Representation\n",
+    "While ONNX models are directly supported by OpenVINO™, it can be useful to convert them to IR format to take advantage of OpenVINO optimization tools and features.\n",
```

Comment:
```suggestion
    "While ONNX models are directly supported by OpenVINO runtime, it can be useful to convert them to IR format to take advantage of OpenVINO optimization tools and features.\n",
```